### ISSUE TITLE
Microsoft broke my config

### ISSUE THREAD

**Issue by cowingtonpost1** [marlam/mpop-mirror#14]
My microsoft email setup using mutt_oauth2.py and the xoauth auth method in mpop worked up until July 6th, when it broke. mpop: POP3 server does not support authentication method XOAUTH2

I have tried letting mpop automatically select an auth method but it rejects my auth token. I don't know how to configure it to make microsoft auth work.

**Comment by marlam** [comment]
Try `mpop --serverinfo` to check which methods the POP3 server supports now. Maybe OAUTHBEARER is one of them, and maybe it works with the same token you used before.

**Comment by cowingtonpost1** [comment]
The server only supports user and plain. I have tried both with my same
auth token and username and it does not work.

**Comment by michaelcadilhac** [comment]
This is due to this commit:
https://git.marlam.de/gitweb/?p=mpop.git;a=commitdiff;h=40feacd396e95e3e878ea423ac2305c98c95b5a2

The condition for "end of token" is not correct.  It should read:
```
           if (string[i + token_len] == ' '
                    || string[i + token_len] == '\r'
                    || string[i + token_len] == '\0') /* valid end of token */
```

**Comment by marlam** [comment]
You are right, I messed up the "end of token" condition. Thank you very much for debugging this.

I wrote this code originally for msmtp, where the `\r\n` at the end of the server reply is always removed before token_in_string() is called, so that the function works correctly there.

I applied your change (commit 9063e68), and will release version 1.4.20 now to get this important bug fix out.

Sorry for the trouble this has caused.
